src/entry.c

```diff
@@ -1483,3 +1484,113 @@ void
 unlock_entry( const struct stumpless_entry *entry ) {
   config_unlock_mutex( entry->mutex );
 }
+
+
+/*
+ * formating entry_to_string values
+ */
+static size_t append_key_value_pair(char *dest, size_t offset, const char *key, const char *value, size_t value_length) {
+    size_t key_len = strlen(key);
+
+    // key="
+    memcpy(dest + offset, key, key_len);
+    offset += key_len;
+    dest[offset++] = '=';
+    dest[offset++] = '"';
+
+    // value
+    if (value && value_length > 0) {
+        memcpy(dest + offset, value, value_length);
+        offset += value_length;
+    }
+
+    // "
+    dest[offset++] = '"';
+    dest[offset++] = ',';
+
+    return offset;
+}
+
+char *stumpless_entry_to_string(const struct stumpless_entry *entry) {
+    if (!entry) {
+        return NULL;
+    }
+
```

Comment:
You'll need to obtain the lock on the entry before gathering these values (and be sure to release it at the end as well). Otherwise, this will not be thread-safe code.